src/firefly/js/templates/lightcurve/PeriodogramOptionsPanel.jsx

```diff
@@ -183,44 +242,65 @@ var periodogramRangeReducer= function(inFields, action) {
 function onSearchSubmit(request) {
     console.log(request);
     //if (request.Tabs==='periodfinding') {
-        doPeriodFinding(request);
+    doPeriodFinding(request);
     //}
     //else {
     //    console.log('request no supported');
     //}
 }
 
 function doPeriodFinding(request) {
-    //let tbl = getTblById(RAW_TABLE);
-    console.log(request);
+    let tbl = getTblById(RAW_TABLE);
+    //let catname0 = get(FieldGroupUtils.getGroupFields(gkey), 'cattable.value', catTable[0].value);
+    const fields = FieldGroupUtils.getGroupFields(gkey);
+    const srcFile = tbl.request.source;
+    console.log(fields);
 
-    var tReq2 = makeTblRequest('LightCurveProcessor',PEAK_TABLE , {
-        'result_table': 'http://web.ipac.caltech.edu/staff/ejoliet/demo/vo-nexsci-result-sample.xml', //For now return result table for non-existing API
+    var tReq2 = makeTblRequest('LightCurveProcessor', PEAK_TABLE, {
+        'original_table': srcFile,
+        'x': fields.x.value || 'mjd',
+        'y': fields.y.value || 'w1mpro_ep',
+        'alg': fields.periodAlgor.value,
+        'pmin': fields.periodMin.value,
+        'pmax': fields.periodMax.value,
+        'step_method': fields.stepMethod.value,
+        'step_size': fields.stepSize.value,
+        'peaks': fields.peaks.value,
+        //'result_table': 'http://web.ipac.caltech.edu/staff/ejoliet/demo/vo-nexsci-result-sample.xml', //For now return result table for non-existing API
         'table_name': PEAK_TABLE
-    },  {tbl_id:PEAK_TABLE});
+    }, {tbl_id: PEAK_TABLE});
 
     if (tReq2 != null) {
         let xyPlotParams = {
             x: {columnOrExpr: 'Peak', options: 'grid'},
             y: {columnOrExpr: 'Power', options: 'grid'}
         };
-        loadXYPlot({chartId:PEAK_TABLE, tblId:PEAK_TABLE, xyPlotParams});
+        loadXYPlot({chartId: PEAK_TABLE, tblId: PEAK_TABLE, xyPlotParams});
         dispatchTableSearch(tReq2, {removable: true});
     }
 
-    var tReq = makeTblRequest('LightCurveProcessor',PERIODOGRAM , {
-        'result_table': 'http://web.ipac.caltech.edu/staff/ejoliet/demo/vo-nexsci-result-sample.xml', //For now return result table for non-existing API
+    var tReq = makeTblRequest('LightCurveProcessor', PERIODOGRAM, {
+        'original_table': srcFile,
+        'x': fields.x.value || 'mjd',
+        'y': fields.y.value || 'w1mpro_ep',
+        'alg': fields.periodAlgor.value,
+        'pmin': fields.periodMin.value,
+        'pmax': fields.periodMax.value,
+        'step_method': fields.stepMethod.value,
+        'step_size': fields.stepSize.value,
+        'peaks': fields.peaks.value,
+        //'result_table': 'http://web.ipac.caltech.edu/staff/ejoliet/demo/vo-nexsci-result-sample.xml', //For now return result table for non-existing API
```

Comment:
I find it easier to read if the prop name is not quoted.  Anything in quotes will be highlighted as string when they are not.